app/src/main/java/com/trevjonez/daggertest/main_activity/MainActivityComponent.java

```diff
@@ -16,18 +16,41 @@
 
 package com.trevjonez.daggertest.main_activity;
 
+import com.trevjonez.daggertest.fragment.MainActivityFragmentBinder;
 import com.trevjonez.inject.PlainComponent;
 import com.trevjonez.inject.activity.ActivityComponentBuilder;
 
+import javax.inject.Named;
+
+import dagger.Module;
+import dagger.Provides;
 import dagger.Subcomponent;
 
 /**
  * @author TrevJonez
  */
-@Subcomponent
+@Subcomponent (modules= {
+        MainActivityFragmentBinder.class,
+        MainActivityComponent.MainActivityModule.class
+})
 @MainActivityScope
 public interface MainActivityComponent extends PlainComponent<MainActivity> {
 
     @Subcomponent.Builder
-    interface Builder extends ActivityComponentBuilder<MainActivity, MainActivityComponent> { }
+    interface Builder extends ActivityComponentBuilder<MainActivity, MainActivityComponent> {
+        Builder activityModule(MainActivityModule module);
+    }
+
+    @Module
+    class MainActivityModule{
```

Comment:
For reusability in this example i separated it. Normally i don't have reusable Modules of Activities, and i prefer Activity dependency codes in one place like this. Same can be done for View and Presenter interfaces like this example: https://www.novoda.com/blog/better-class-naming/